include/CL/sycl/device.hpp

```diff
@@ -245,10 +245,10 @@ class device
   */
   vector_class<device>
   create_sub_devices(info::device_partition_type partition_type,
-                     info::device_partition_property partition_property,
-                     info::device_affinity_domain affinity_domain) const {
+                     info::partition_property device_partition_property,
+                     info::partition_affinity_domain affinity_domain) const {
```

Comment:
Good starting point but I just started these function and commented it out because it does not mean anything for now...

Reply:
And this function also updated in SYCL 1.2.1

It is replaced to three functions below.
// Available only when prop == info::partition_property::partition_equally
template <info::partition_property prop>
vector_class<device> create_sub_devices(size_t nbSubDev) const;
// Available only when prop == info::partition_property::partition_by_counts
template <info::partition_property prop>
vector_class<device> create_sub_devices(const vector_class<size_t> &counts) const;
// Available only when prop == info::partition_property::partition_by_affinity_domain
template <info::partition_property prop>
vector_class<device> create_sub_devices(info::affinity_domain affinityDomain) const;

Should I keep it in XYZTRISYCL_OPENCL but update it?

Reply:
`XYZTRISYCL_OPENCL` was just to skip it to avoid compilation errors but if you have a fixed version, at least just for the function declaration, you can add the functions and remove `XYZTRISYCL_OPENCL`.